### PR TITLE
modernize gem structure; Fixes #158

### LINKED ISSUE #158: Modernize gem file structure
(following the pattern of a new gem skeleton with `bundle gem git-reflow`)
* Add `bin/console`
* make `exe` the directory for executables in the release
* move `bin/git-reflow` to `exe/git-reflow`

### PR DESCRIPTION
**Pull request by pboling**
Fixes #158

### CODE REVIEW

File: Gemfile.lock [hunk 1]
```diff
@@ -104,13 +104,13 @@ PLATFORMS
...
-  bundler
+  bundler (~> 1.11)
   chronic
   git_reflow!
   pry-byebug
-  rake
+  rake (~> 10.0)
   rdoc
-  rspec (~> 3.0.0)
+  rspec (~> 3.0)
```
Comment (pboling) [inline Gemfile.lock]: Note, I tried updating rspec to 3.4, and it fails the spec suite miserably.  Going to require some effort to fix compatibility with rspec 3.4.

File: bin/console [hunk 2]
```diff
@@ -0,0 +1,14 @@
+#!/usr/bin/env ruby
+
+require "bundler/setup"
+require "git_reflow"
+
+# You can add fixtures and/or initialization code here to make experimenting
```
Comment (codenamev) [inline bin/console]: Can we remove these comments?

File: bin/setup [hunk 3]
```diff
@@ -0,0 +1,8 @@
+#!/usr/bin/env bash
+set -euo pipefail
+IFS=$'\n\t'
+set -vx
+
+bundle install
+
+# Do any other automated setup for gem development that you need to do here
```
Comment (codenamev) [inline bin/setup]: Remove comment here too please

**Comment by codenamev** [comment]
I must admit I'm not a fan of bundler's move to the new `exe` executable file pattern as it conflicts with [rubygems standardized patterns](http://guides.rubygems.org/patterns/).  We are using bundler though so we might as well update it.

:+1: once those comments are yanked. Thanks for the contribution :smile:.

**Comment by codenamev** [comment]
<a href="https:&#x2F;&#x2F;trello.com&#x2F;c&#x2F;ELpi2X5A&#x2F;10-modernize-gem-structure"> modernize gem structure</a>

**Comment by pboling** [comment]
The `bin/console` tool is *super* useful for playing with the gem's code, that was the driver for the change.  And it shouldn't be released with the gem.  It is unfortunate that the RubyGems guide and the bundler pattern are out of sync 😢

**Comment by pboling** [comment]
@codenamev comments removed.

**Comment by codenamev** [comment]
Yeah `bin/console` is really neat.  I came across this in the reality gem not too long ago and really liked it.

LGTM, nice work!

**Pull request merged**